x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/enrich_signal_threat_matches.ts

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { get, isObject } from 'lodash';
+
+import type { SignalSearchResponse, SignalSourceHit } from '../types';
+import type {
+  GetMatchedThreats,
+  ThreatIndicator,
+  ThreatListItem,
+  ThreatMatchNamedQuery,
+} from './types';
+import { extractNamedQueries } from './utils';
+
+const DEFAULT_INDICATOR_PATH = 'threat.indicator';
+const getSignalId = (signal: SignalSourceHit): string => signal._id;
+
+export const groupAndMergeSignalMatches = (signalHits: SignalSourceHit[]): SignalSourceHit[] => {
+  const dedupedHitsMap = signalHits.reduce<Map<unknown, SignalSourceHit>>((acc, signalHit) => {
+    const signalId = getSignalId(signalHit);
+    if (!acc.has(signalId)) {
+      acc.set(signalId, signalHit);
+    } else {
+      const existingSignalHit = acc.get(signalId) as SignalSourceHit;
```

Comment:
If you move this line to be above the if statement you can remove the `as` casting by doing the following:

```node.js
const signalId = getSignalId(signalHit);
const existingSignalHit = acc.get(signalId);
if (!existingSignalHit) {
  acc.set(signalId, signalHit);
} else {...
```

Reply:
Good call: b6e176c